### PR TITLE
Add readiness probe for router

### PR DESCRIPTION
**Pull request by menakaj**
### Issues

Fixes https://github.com/wso2/product-microgateway/issues/2102

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/internal/adapter/adapter.go [hunk 1]
```diff
@@ -312,4 +312,7 @@ func fetchAPIsOnStartUp(conf *config.Config) {
 			}(data)
 		}
 	}
+	// All apis are fected. Deploy the /ready route for the readiness and startup probes.
```
Comment (Rajith90) [inline adapter/internal/adapter/adapter.go:315]:
```suggestion
	// All apis are fetched. Deploy the /ready route for the readiness and startup probes.
```

File: adapter/internal/adapter/adapter.go [hunk 2]
```diff
@@ -312,4 +312,7 @@ func fetchAPIsOnStartUp(conf *config.Config) {
...
+	xds.DeployReadinessAPI(envs)
+	logger.LoggerMgw.Info("Fetching APIs completed...")
```
Comment (Rajith90) [inline adapter/internal/adapter/adapter.go:317]: Its better if we can print the number of APIs fetched as well

File: adapter/internal/oasparser/envoyconf/constants.go [hunk 3]
```diff
@@ -79,11 +79,13 @@ const (
 const (
 	healthPath  string = "/health"
 	testKeyPath string = "/testkey"
+	readyPath   string = "/ready"
```
Comment (Rajith90) [inline adapter/internal/oasparser/envoyconf/constants.go:82]: Please modify the k8s artefacts accordingly as well

File: adapter/internal/adapter/adapter.go (hunk 2 above)
Comment (menakaj) [reply adapter/internal/adapter/adapter.go:317]:
```
2021-06-27 17:32:38 INFO [apis_fetcher.go:249] - [synchronizer.PushAPIProjects] [-] Successfully deployed 3 APIs
2021-06-27 17:32:38 INFO [server.go:224] - [xds.DeployReadinessAPI] [-] Finished fetching APIs from the Control Plane. Deploying the readiness endpoint...
```

**Review by pubudu538 — APPROVED** [review]

File: adapter/internal/oasparser/envoyconf/constants.go (hunk 3 above)
Comment (menakaj) [reply adapter/internal/oasparser/envoyconf/constants.go:82]: Fixed in https://github.com/wso2/product-microgateway/pull/2101/commits/b1549c07886f6a498d0323b293f922bc4fe2b3d8

File: adapter/internal/adapter/adapter.go (hunk 2 above)
Comment (menakaj) [reply adapter/internal/adapter/adapter.go]: Fixed in https://github.com/wso2/product-microgateway/pull/2101/commits/f8b1a0f26d76982947e760b92ce30641bbf7e936

File: adapter/internal/adapter/adapter.go (hunk 1 above)
Comment (menakaj) [reply adapter/internal/adapter/adapter.go]: Fixed in https://github.com/wso2/product-microgateway/pull/2101/commits/f8b1a0f26d76982947e760b92ce30641bbf7e936

**Pull request merged**